### PR TITLE
added implementation of login endpoint

### PR DESCRIPTION
**Pull request by nastasiiaKulinich**

### CODE REVIEW

File: src/main/java/com/example/bookstore/config/SecurityConfig.java [hunk 1]
```diff
@@ -1,22 +1,30 @@
...
 import org.springframework.security.core.userdetails.UserDetailsService;
 import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
 import org.springframework.security.crypto.password.PasswordEncoder;
 import org.springframework.security.web.SecurityFilterChain;
+import org.springframework.security.web.authentication.UsernamePasswordAuthenticationFilter;
 
 @EnableMethodSecurity
 @RequiredArgsConstructor
 @Configuration
+@EnableWebSecurity
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/bookstore/config/SecurityConfig.java]: Do you need both of these annotations `@EnableMethodSecurity and @EnableWebSecurity`?

File: src/main/java/com/example/bookstore/security/JwtAuthenticationFilter.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+            Authentication authentication = new UsernamePasswordAuthenticationToken(
+                    userDetails, null, userDetails.getAuthorities()
+            );
+            SecurityContextHolder.getContext().setAuthentication(authentication);
+        }
+        filterChain.doFilter(request, response);
+    }
+
+    private String getToken(HttpServletRequest request) {
+        String bearerToken = request.getHeader("Authorization");
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/bookstore/security/JwtAuthenticationFilter.java]: Use Spring constant `HttpHeaders.AUTHORIZATION`

File: src/main/java/com/example/bookstore/security/JwtAuthenticationFilter.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+        if (StringUtils.hasText(bearerToken) && bearerToken.startsWith("Bearer ")) {
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/bookstore/security/JwtAuthenticationFilter.java]: create constant for "Bearer "

File: src/main/java/com/example/bookstore/service/book/BookServiceImpl.java [hunk 4]
```diff
@@ -60,9 +59,9 @@ public void deleteById(Long id) {
     @Override
     public List<BookDto> search(BookSearchParametersDto searchParametersDto) {
         Specification<Book> bookSpecification = bookSpecificationBuilder.build(searchParametersDto);
-        return bookRepository.findAll((Sort) bookSpecification)
+        return bookRepository.findAll(bookSpecification)
                 .stream()
-                .map(bookMapper::toDto)
+                .map((Object book) -> bookMapper.toDto((Book) book))
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/bookstore/service/book/BookServiceImpl.java:64]: Why do you need this changes?

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
see pr

File: src/main/java/com/example/bookstore/config/SecurityConfig.java (hunk 1 above)
Comment (nastasiiaKulinich) [reply src/main/java/com/example/bookstore/config/SecurityConfig.java]: No, I'll remove the second annotation

File: src/main/java/com/example/bookstore/service/book/BookServiceImpl.java (hunk 4 above)
Comment (nastasiiaKulinich) [reply src/main/java/com/example/bookstore/service/book/BookServiceImpl.java:64]: I found out that search endpoint doesn't work with previous code, it broke at the moment of casting to (Sort)

File: src/main/java/com/example/bookstore/security/JwtAuthenticationFilter.java [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+            );
+            SecurityContextHolder.getContext().setAuthentication(authentication);
+        }
+        filterChain.doFilter(request, response);
+    }
+
+    private String getToken(HttpServletRequest request) {
+        String bearerToken = request.getHeader(HttpHeaders.AUTHORIZATION);
+        if (StringUtils.hasText(bearerToken) && bearerToken.startsWith(BEARER_PREFIX)) {
+            return bearerToken.substring(7);
```
Comment (okuzan) [inline src/main/java/com/example/bookstore/security/JwtAuthenticationFilter.java]:
```suggestion
            return bearerToken.substring(BEARER.length());
```
they are logically connected, and good code should ideally reflect this

**Review by okuzan — CHANGES_REQUESTED** [review]

**Review by Elena-Bruyako — APPROVED** [review]

**Pull request merged**